app/app.rb

```diff
@@ -0,0 +1,75 @@
+ENV['RACK_ENV'] ||= 'development'
+
+require 'sinatra/base'
+require_relative 'datamapper_setup'
+
+class Chitter < Sinatra::Base
+  enable :sessions
+  set :session_secret, 'super secret'
+
+  get '/' do
+    if current_user.nil?
+      erb :'/sign_up_and_log_in/user_sign_up'
+    else
+      redirect('/peeps')
+    end
+  end
+
+  post '/signup' do
+    users = User.all
+    if !(User.get(email: params[:email]).nil?) || !(User.get(username: params[:username]).nil?)
```

Comment:
Metrics/LineLength: Line is too long. [95/90] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)